Hide `HoldForMenuButton` on non-touch devices

rushiiMachine opened the pull request:
As per [this](https://discord.com/channels/188630481301012481/188630652340404224/1187604735574954014) discussion in the discord, this PR changes the `HoldForMenuButton` to never appear on non-touch devices until the cursor is nearby, and on touch devices it will retain the pop-in animation, fading out to always be slightly visible. Since this component is redundant on devices on devices with a keyboard and hardcoded regardless it should be at least hidden.

rushiiMachine committed:
Hide `HoldForMenuButton` on non-touch devices

Disable the enter animation and fully hide until cursor is near on non-touch devices.

rushiiMachine commented on osu.Game/Screens/Play/HUD/HoldForMenuButton.cs:
```diff
@@ -99,9 +111,11 @@ protected override void Update()
                 Alpha = 1;
             else
             {
+                float minAlpha = AutoHide.Value ? .001f : .07f;
```
I tried allowing the alpha to clamp to 0, (and enabling `AlwaysPresent` on the container), however ran into issues with testing as the interpolation below seemed to never reach a fixed value (and `IsPresent` is useless when `AlwaysPresent` is enabled`)

peppy commented on osu.Game/Screens/Play/HUDOverlay.cs:
```diff
@@ -158,7 +158,7 @@ public HUDOverlay([CanBeNull] DrawableRuleset drawableRuleset, IReadOnlyList<Mod
                     Children = new Drawable[]
                     {
-                        HoldToQuit = CreateHoldForMenuButton(),
+                        HoldToQuit = CreateHoldForMenuButton(touchDevice: mods.Any(m => m is ModTouchDevice)),
```
Rather than doing this, use `SessionStatics` from within the class, don't pass anything in.

https://github.com/ppy/osu/blob/63a0ea54109c93984501b66435febf676663cab4/osu.Game/Screens/Play/PlayerTouchInputDetector.cs#L24-L29

peppy replied:
Please don't self resolve issues. This isn't even fixed.

rushiiMachine replied:
I misunderstood it as "dont pass anything to `CreateHoldForMenuButton`", I assume you meant inject it from within `HoldForMenuButton`?

rushiiMachine committed:
Invert `HoldForMenuButton#AutoHide` and use `SessionStatics`

Use SessionStatics to supply touch device data and invert the autohide parameter to better fit the supplied Bindable.

peppy commented:
I've taken part of what you did here and PR'd it separately with some simplification: https://github.com/ppy/osu/pull/26542.
